### PR TITLE
Fill out and fix some node interfaces 

### PR DESCRIPTION
**Pull request by henridf**
I'm curious to know if there is any way to avoid the redundancy described in the commit message of 0f574c0.

### CODE REVIEW

**Comment by henridf** [comment]
#1137

File: lib/node.js [hunk 1]
```diff
@@ -353,9 +353,11 @@ declare module "crypto" {
   ): void;
 }
 
+type address = {address: string; family: string; port: number};
```
Comment (mroch) [inline lib/node.js]: everything in libs is global, so we need to give this a more specific name. `dgram$Socket$address`, perhaps?

File: lib/node.js [hunk 2]
```diff
@@ -687,7 +698,28 @@ declare module "http" {
...
+    listen(handle: Object, callback?: Function): Server;
+    close(callback?: Function): Server;
+    setTimeout(msecs: number, callback: Function): Server;
+    timeout: number;
+  }
+
+  declare class ClientRequest extends http$ClientRequest {}
+  declare class IncomingMessage extends http$IncomingMessage {}
+
+  declare function createServer(options: Object, secureConnectionListener?: function): typeof Server;
```
Comment (mroch) [inline lib/node.js]: `typeof Server` returns the `Server` class, not an instance of `Server`. I think you want to drop the `typeof`?

File: lib/node.js (hunk 1 above)
Comment (henridf) [inline lib/node.js]: ah yes, fixed

File: lib/node.js (hunk 2 above)
Comment (henridf) [inline lib/node.js]: Fixed. 

I was a little fuzzy on the meaning of `typeof` and did this by imitation of other places (e.g. `tls.Server.createServer`) that appear to be making the same error.  
(I'll try to make a pass to fix in a separate PR at some point).

**Comment by mroch** [comment]
@facebook-github-bot shipit

**Pull request closed without merging**